core/lib/include/kernel_defines.h

```diff
@@ -280,6 +280,61 @@ extern "C" {
 
 /* end{code-style-ignore} */
 
+/** @brief Declare a function as deprecated
+ *
```

Comment:
```suggestion
 *
 * @param reason    Reason for removal
 * @param eol       Expected end of life
 * @param migration Migration path (e.g. API to use instead)
 *
```

Maybe enforce consistency and completeness of the doc by having explicit arguments for the important details. That would also decrease the itch to use fancy formatting.

Beware: Indent levels are not correct, did this on a phone.